pallets/loans/src/lib.rs

```diff
@@ -515,6 +543,30 @@ pub mod module {
             )?;
             Ok(().into())
         }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn add_reserves(
+            origin: OriginFor<T>,
+            currency_id: CurrencyId,
+            add_amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            let root = ensure_signed(origin)?;
+            Self::add_reserves_internal(root, currency_id, add_amount)?;
```

Comment:
There is no need to add an internal function for every dispatchable call. The main point of the internal function is it can be reused by other calls.

Reply:
Got it. I will modify it.